docs/advanced/document-store/index.md

```diff
@@ -346,6 +346,179 @@ array([[7., 7., 7.],
+However, when we try to create another view of the DocumentArray in `da2`, we get a fresh DocumentArray.
+
+We also expect the script to increment the length of the DocumentArrays every time we run it.
+This is because the previous run should have saved the length of the DocumentArray with `index_name="my_index"` and our most recent run will append a new document, incrementing the length by `+1` each time.
+
+However, it seems like our append operation is also not being persisted.
+
+````{dropdown} What actually happened here?
+The DocumentArray actually did persist.
+but not in the way we might expect.
+Since we did not use the `with` context manager or scope our mutation, the persistence logic is being evaluated when the program exits.
+`da1` is destroyed first, persisting the DocumentArray of length `1`.
+But when `da2` is destroyed, it persists a DocumentArray of length `0` to the same index in Redis as `da1`, overriding its value.
+
+This means that if we had not created `da2`, the overriding would not have occured and the script would actually increment the length of the DocumentArray correctly.
+You can prove this to yourself by commenting out the last 2 lines of the script and running the script repeatedly.
+
+**Script**
+```python
+from docarray import DocumentArray, Document
+
+da1 = DocumentArray(storage='redis', config=dict(n_dim=3, index_name="my_index"))
+da1.append(Document())
+print(f"Length of da1 is {len(da1)}")
+
+# da2 = DocumentArray(storage='redis', config=dict(n_dim=3, index_name="my_index"))
+# print(f"Length of da2 is {len(da2)}")
+```
+
+**First run output**
+```console
+Length of da1 is 1
+```
+**Second run output**
+```console
+Length of da1 is 2
+```
+**Third run output**
+```console
+Length of da1 is 3
+```
+````
+
+Now that we know the issue, let's explore some idioms we can use to work with DocumentArrays backed by external storage in a more predictable manner.
+### Using Context Manager
+The recommended way is to use the DocumentArray as a context manager like so:
+
+```python
+from docarray import DocumentArray, Document
+
+da1 = DocumentArray(storage='redis', config=dict(n_dim=3, index_name="my_index"))
+with da1:  # Let's use the context manager to make sure we persist the mutation
+    da1.append(Document())  #
+print(f"Length of da1 is {len(da1)}")
+
+da2 = DocumentArray(storage='redis', config=dict(n_dim=3, index_name="my_index"))
+print(f"Length of da2 is {len(da2)}")
+```
+**First run output**
+```console
+Length of da1 is 1
+Length of da2 is 1
+```
+**Second run output**
+```console
+Length of da1 is 2
+Length of da2 is 2
+```
+**Third run output**
+```console
+Length of da1 is 3
+Length of da2 is 3
+```
+
+The append we made to the DocumentArray is now persisted properly. Hurray!
+
+### Using Scope
+Another method that is sometimes useful when multiple DocumentArrays are involed is to perform the mutation in a function scope.
```

Comment:
Honestly I don't think that we should even tell users about this.
Using the context manager seems safer and more predictable, and it would be our single (official) way of doing things, which is always good.

Reply:
More than that actually I am not sure that the behavior will persist indefinitely (with future releases). And users should never do that so I would agree not to tell about that part.

Reply:
Yup agree. Seems like bad practice so probably shouldn't recommend it though it works in the current release.

Reply:
Should we not support it then? I think @alaeddine-13 is proposing we don't have the persistence logic in `__del__` at all in his [latest comment on the SystemExit behavior PR](https://github.com/jina-ai/docarray/pull/604#pullrequestreview-1138612018)

Reply:
No function scope at all, this should be removed. Plus in Python u do not know when they will be released